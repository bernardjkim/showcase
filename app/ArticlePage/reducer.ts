/*
 *
 * ArticlePage reducer
 *
 */

import { ArticleActionTypes, ArticleState, SET_ARTICLE_ID } from './types';

// The initial state of the App
export const initialState: ArticleState = {
  articleId: undefined,
};

function articlePageReducer(state: ArticleState = initialState, action: ArticleActionTypes) {
  switch (action.type) {
    case SET_ARTICLE_ID:
      return {
        ...state,
        articleId: action.articleId,
      };

    default:
      return state;
  }
}

export default articlePageReducer;
