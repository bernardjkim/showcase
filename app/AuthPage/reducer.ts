/*
 *
 * AuthPage reducer
 *
 */

// import { LOCATION_CHANGE } from 'react-router-redux';
import {
  AuthState,
  AuthActionTypes,
  CLEAR_ERRORS,
  CREATE_TOKEN,
  CREATE_TOKEN_ERROR,
  CREATE_TOKEN_SUCCESS,
  CREATE_USER,
  CREATE_USER_ERROR,
  CREATE_USER_SUCCESS,
} from './types';

export const initialState: AuthState = {
  loading: false,
  error: undefined,
};

function AuthPageReducer(
  state: AuthState = initialState,
  action: AuthActionTypes,
) {
  switch (action.type) {
    // case LOCATION_CHANGE:
    //   return { ...state, error: undefined };

    case CLEAR_ERRORS:
      return { ...AbortController, error: undefined };

    case CREATE_TOKEN:
      return { ...state, loading: true, error: undefined };

    case CREATE_TOKEN_SUCCESS:
      return { ...state, loading: false };

    case CREATE_TOKEN_ERROR: {
      return { ...state, loading: false, error: { createToken: action.error } };
    }

    case CREATE_USER:
      return { ...state, loading: true, error: undefined };

    case CREATE_USER_SUCCESS:
      return { ...state, loading: false };

    case CREATE_USER_ERROR:
      return { ...state, loading: false, error: { createUser: action.error } };

    default:
      return state;
  }
}

export default AuthPageReducer;
