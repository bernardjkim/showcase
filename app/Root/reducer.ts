/*
 *
 * App reducer
 *
 */

import {
  DELETE_TOKEN,
  DELETE_TOKEN_ERROR,
  DELETE_TOKEN_SUCCESS,
  LOAD_USER,
  LOAD_USER_ERROR,
  LOAD_USER_SUCCESS,
  RootActionTypes,
  RootState,
  SET_USER,
} from './types';

// The initial state of the App
export const initialState: RootState = {
  user: undefined,
};

function AppReducer(state: RootState = initialState, action: RootActionTypes) {
  switch (action.type) {
    case SET_USER:
      return { ...state, user: action.user };

    case DELETE_TOKEN:
      return { ...state, loading: true, error: undefined };

    case DELETE_TOKEN_SUCCESS:
      return { ...state, loading: false };

    case DELETE_TOKEN_ERROR:
      return { ...state, loading: false, error: action.error };

    case LOAD_USER:
      return { ...state, user: undefined, validateToken: false, loading: true, error: undefined };

    case LOAD_USER_SUCCESS:
      return { ...state, user: action.user ? action.user : undefined, loading: false };

    case LOAD_USER_ERROR:
      return { ...state, loading: false, error: { loadUser: action.error } };

    default:
      return state;
  }
}

export default AppReducer;
