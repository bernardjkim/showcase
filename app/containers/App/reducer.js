/*
 * 
 * App reducer
 * 
 */

import { fromJS } from 'immutable';
import {
  CREATE_TOKEN,
  CREATE_TOKEN_ERROR,
  CREATE_TOKEN_SUCCESS,
} from './constants';

// The initial state of the App
export const initialState = fromJS({
  loading: false,
  error: false,
  token: false,
});

function appReducer(state = initialState, action) {
  switch (action.type) {
    case CREATE_TOKEN:
      return state.set('loading', true).set('error', false);

    case CREATE_TOKEN_SUCCESS:
      return state.set('token', action.token).set('loading', false);

    case CREATE_TOKEN_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    default:
      return state;
  }
}

export default appReducer;
