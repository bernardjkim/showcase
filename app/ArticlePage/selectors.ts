import { createSelector } from 'reselect';
import { initialState } from './reducer';
import { ArticleState } from './types';

/**
 * Direct selector to the articlePage state domain
 */

const selectArticlePageDomain = (state: any): ArticleState => state.get('articlePage', initialState);

/**
 * Other specific selectors
 */

const makeSelectLoading = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.loading,
  );

const makeSelectError = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.error,
  );

const makeSelectArticle = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.article,
  );

const makeSelectComments = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.comments,
  );

const makeSelectLikes = () =>
  createSelector(
    selectArticlePageDomain,
    articleState => articleState.likes,
  );
/**
 * Default selector used by ArticlePage
 */

const makeSelectArticlePage = () =>
  createSelector(
    selectArticlePageDomain,
    substate => substate,
  );

export default makeSelectArticlePage;
export {
  selectArticlePageDomain,
  makeSelectArticle,
  makeSelectComments,
  makeSelectLikes,
  makeSelectLoading,
  makeSelectError,
};
