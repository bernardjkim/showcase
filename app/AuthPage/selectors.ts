import { createSelector } from 'reselect';
import { initialState } from './reducer';
import { AuthState } from './types';

/**
 * Direct selector to the AuthPage state domain
 */

const selectAuthPageDomain = (state: any): AuthState =>
  state.get ? state.get('authPage', initialState) : initialState;

/**
 * Other specific selectors
 */
const makeSelectLoading = () =>
  createSelector(
    selectAuthPageDomain,
    authState => authState.loading,
  );

const makeSelectError = () =>
  createSelector(
    selectAuthPageDomain,
    authState => authState.error,
  );

/**
 * Default selector used by AuthPage
 */

const makeSelectAuthPage = () =>
  createSelector(
    selectAuthPageDomain,
    substate => substate,
  );

export default makeSelectAuthPage;
export { selectAuthPageDomain, makeSelectError, makeSelectLoading };
