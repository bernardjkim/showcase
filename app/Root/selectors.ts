import { createSelector } from 'reselect';
import { initialState } from './reducer';
import { RootState } from './types';

const selectRouter = (state: any) => state.get('router');

const makeSelectLocation = () =>
  createSelector(
    selectRouter,
    routerState => routerState.get('location').toJS(),
  );

/**
 * Direct selector to the global state domain
 */

const selectGlobalDomain = (state: any): RootState => (state.get ? state.get('global', initialState) : initialState);

/**
 * Other specific selectors
 */

const makeSelectLoading = () =>
  createSelector(
    selectGlobalDomain,
    globalState => globalState.loading,
  );

const makeSelectError = () =>
  createSelector(
    selectGlobalDomain,
    globalState => globalState.error,
  );

const makeSelectUser = () =>
  createSelector(
    selectGlobalDomain,
    globalState => globalState.user,
  );

const makeSelectValidateToken = () =>
  createSelector(
    selectGlobalDomain,
    globalState => globalState.validateToken,
  );

/**
 * Default selector used by App
 */

const makeSelectApp = () =>
  createSelector(
    selectGlobalDomain,
    substate => substate,
  );

export default makeSelectApp;
export {
  makeSelectLocation,
  selectGlobalDomain,
  makeSelectUser,
  makeSelectValidateToken,
  makeSelectLoading,
  makeSelectError,
};
