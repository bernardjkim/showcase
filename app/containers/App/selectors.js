import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectRouter = state => state.get('router');

const makeSelectLocation = () =>
  createSelector(selectRouter, routerState =>
    routerState.get('location').toJS(),
  );

/**
 * Direct selector to the global state domain
 */

const selectGlobalDomain = state => state.get('global', initialState);

/**
 * Other specific selectors
 */

const makeSelectLoading = () =>
  createSelector(selectGlobalDomain, globalState => globalState.get('loading'));

const makeSelectError = () =>
  createSelector(selectGlobalDomain, globalState => globalState.get('error'));

const makeSelectUser = () =>
  createSelector(selectGlobalDomain, globalState => globalState.get('user'));

const makeSelectValidateToken = () =>
  createSelector(selectGlobalDomain, globalState =>
    globalState.get('validateToken'),
  );

/**
 * Default selector used by App
 */

const makeSelectApp = () =>
  createSelector(selectGlobalDomain, substate => substate.toJS());

export default makeSelectApp;
export {
  makeSelectLocation,
  selectGlobalDomain,
  makeSelectUser,
  makeSelectValidateToken,
  makeSelectLoading,
  makeSelectError,
};
