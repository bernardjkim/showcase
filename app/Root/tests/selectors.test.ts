import { fromJS } from 'immutable';

import { initialState } from '../reducer';
import { makeSelectLocation, makeSelectUser, selectGlobalDomain } from '../selectors';

describe('makeSelectLocation', () => {
  const locationStateSelector = makeSelectLocation();
  it('should select the location', () => {
    const mockedState = fromJS({
      router: { location: { pathname: '/foo' } },
    });
    expect(locationStateSelector(mockedState)).toEqual(mockedState.getIn(['router', 'location']).toJS());
  });
});

describe('selectGlobalDomain', () => {
  it('should select the global state', () => {
    expect(selectGlobalDomain(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectUser', () => {
  const userSelector = makeSelectUser();
  it('should select the user state', () => {
    expect(userSelector(initialState)).toMatchSnapshot();
  });
});
