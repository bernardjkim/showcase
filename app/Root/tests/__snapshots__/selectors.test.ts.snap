// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeSelectError should select the error state 1`] = `undefined`;

exports[`makeSelectLoading should select the loading state 1`] = `false`;

exports[`makeSelectUser should select the user state 1`] = `undefined`;

exports[`makeSelectValidateToken should select the token state 1`] = `true`;

exports[`selectGlobalDomain should select the global state 1`] = `
Object {
  "error": undefined,
  "loading": false,
  "user": undefined,
  "validateToken": true,
}
`;
